### PR TITLE
Desing and backend of the section of seller user ready

### PR DESCRIPTION
**Pull request by IsmaelMontiel710**
# Title: Implementation of Car CRUD with Image Support and Dashboard Enhancements for Sellers
@MeganPegueros  worked here
 #11 #21  #23 
## Description:

This pull request introduces a comprehensive set of features aimed at enhancing the user experience for both car sellers and platform administrators. The changes are centered around the implementation of a full CRUD (_Create, Read, Update, Delete_) system for managing cars, the integration of image support for better visual representation of listings, and the design of several key sections to improve the data visibility for sellers.

---

## Key features and updates:

1. **CRUD Operations for Cars:**
   - Implemented a complete CRUD system for car listings, enabling users to create, read, update, and delete car entries.
   - Each car entry now contains detailed information such as make, model, year, price, and additional metadata required for both user and seller convenience.
   - Integrated form validation for car details to ensure data consistency and prevent erroneous or incomplete entries.

2. **Image Upload and Management:**
   - Developed functionality for sellers to upload images for each car listing. Multiple images can be attached to a single listing, providing a better visual experience for potential buyers.
   - Images are stored efficiently, with handling for different image formats, size restrictions, and automatic resizing to optimize platform performance.

3. **Seller Dashboard Enhancements:**
   - Created a new section in the seller’s dashboard that provides key metrics related to the seller’s car listings.
   - This includes a breakdown of:
     - **Total number of cars listed**
     - **Number of cars sold**
     - **Number of cars that remain unsold**
   - Implemented real-time data updates to reflect the current status of each car (_sold or unsold_) as transactions are completed.
   - Added filtering and sorting options for sellers to easily manage their inventory by various criteria such as price, make, or status.

4. **User Interface and Design:**
   - Designed an intuitive interface for managing car listings, including the addition of buttons for adding new cars, editing existing listings, and viewing detailed information about each car.
   - Ensured responsive design across all screens to support mobile and desktop views.
   - Improved accessibility features by ensuring that all buttons, inputs, and labels are clearly marked and easily navigable for all users.

5. **Error Handling and Feedback:**
   - Incorporated comprehensive error messages and feedback mechanisms, both for users and sellers, to notify them of any issues during CRUD operations (e.g., failed image upload, missing fields, etc.).
   - Ensured that all operations are performed efficiently and handled gracefully in case of connectivity issues or server errors.

---

## Conclusion:

This pull request delivers significant improvements to the platform by offering sellers the tools to manage their car listings with greater ease, including the ability to upload images and track inventory status. The implementation of a robust CRUD system ensures data consistency and user-friendly interactions. With the enhanced dashboard, sellers can now monitor their sales performance in real time, contributing to a more effective and enjoyable selling experience.

image
image
image
image
image
image
image
image
image
image

### CODE REVIEW

File: .gradle/8.7/executionHistory/executionHistory.lock
Comment (Angel591Diaz) [inline .gradle/8.7/executionHistory/executionHistory.lock:1]: uploading unnecessary files

**Review by MeganPegueros — APPROVED** [review]
Very good

**Review by Kamila04 — APPROVED** [review]
Good 👍

File: app/build.gradle.kts
Comment (Angel591Diaz) [inline app/build.gradle.kts:1]: unnecessary dependencies

File: app/src/main/AndroidManifest.xml
Comment (Angel591Diaz) [inline app/src/main/AndroidManifest.xml:1]: ?

File: app/src/main/res/layout/dialog_confirm_delete.xml
Comment (Angel591Diaz) [inline app/src/main/res/layout/dialog_confirm_delete.xml:1]: change the text to english

File: app/src/main/res/layout/dialog_edit_car.xml
Comment (Angel591Diaz) [inline app/src/main/res/layout/dialog_edit_car.xml:1]: Those comments in Spanish? O.o

File: local.properties
Comment (Angel591Diaz) [inline local.properties:1]: a

**Review by Angel591Diaz — APPROVED** [review]
I accept it because it is Mariana's first pr. ;)

**Pull request merged**